test/integration/webhook/operatingsystemconfig/operatingsystemconfig_test.go

```diff
@@ -0,0 +1,184 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package operatingsystemconfig_test
+
+import (
+	"bytes"
+	_ "embed"
+	"encoding/json"
+	"text/template"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener-extension-shoot-rsyslog-relp/pkg/apis/rsyslog"
+	testcommon "github.com/gardener/gardener-extension-shoot-rsyslog-relp/test/common"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+)
+
+type configData struct {
+	// ProjectName to fill in template for config.
+	ProjectName string
+	// Target to fill in template for config.
+	Target string
+	// Port to fill in template for config.
+	Port int
+}
+
+var _ = Describe("Webhook tests", func() {
+	var (
+		osc        *extensionsv1alpha1.OperatingSystemConfig
+		config     *rsyslog.RsyslogRelpConfig
```

Comment:
Why we use the internal type (`import "github.com/gardener/gardener-extension-shoot-rsyslog-relp/pkg/apis/rsyslog"`) and not the external one (`import "github.com/gardener/gardener-extension-shoot-rsyslog-relp/pkg/apis/rsyslog/v1alpha1"`)?